Introduce mesh dataset as possible elevation terrain (#5470)
qgis/QGIS-Documentation: docs/user_manual/introduction/qgis_gui.rst

    * an :guilabel:`Online` service, loading `elevation tiles
      <http://s3.amazonaws.com/elevation-tiles-prod/terrarium/{z}/{x}/{y}.png>`_
      produced by Mapzen tools -- more details at https://registry.opendata.aws/terrain-tiles/
    * a loaded :guilabel:`Mesh` dataset
  * :guilabel:`Elevation`: Raster or mesh layer to be used for generation of
    the terrain.
    The raster layer must contain a band that represents elevation.

>     The mesh layer will use the Z values of its vertices.

Reviewer: ```suggestion
    For a mesh layer, the Z values of the vertices are used.
```
